pinot-core/src/main/java/com/linkedin/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -70,20 +70,23 @@ public InstancePlanMakerImplV2(QueryExecutorConfig queryExecutorConfig) {
 
   @Override
   public PlanNode makeInnerSegmentPlan(IndexSegment indexSegment, BrokerRequest brokerRequest) {
+    BrokerRequest copiedBrokerRequest = brokerRequest.deepCopy();
```

Comment:
This will deepCopy the original request for all segments. That can be costly as we scale for segments.

Aggregation[GroupBy]Plan node can rewrite instead of blindly adding the suffix as it does right now. Avoids the cost of copying full request (which can be large for large queries with big IN clause)